src/kademlia/node.cpp

```diff
@@ -1118,6 +1135,30 @@ void node::incoming_request(msg const& m, entry& e)
 				, reply);
 		}
 	}
+	else if (query == "sample_infohashes")
+	{
+		key_desc_t const msg_desc[] = {
+			{"target", bdecode_node::string_t, 20, 0},
+			{"want", bdecode_node::list_t, 0, key_desc_t::optional},
+		};
```

Comment:
can you think of a way to test this without having to set up a full simulation of the DHT? I suppose you could have the test send this message directly to the dht node